src/glang/lib/std/type_traits.c3

```diff
@@ -23,12 +23,11 @@ typedef [V] AreEquivalent<V, V> : TrueType
 typedef [U, V] SelectFirst : U
 typedef [Type, Condition] TypeIf : SelectFirst<Type, EnableIf<Condition>>
 
-typedef [T] Ref : T&
-typedef [T] Ref<T&> : T&
-typedef [T] Ref<T mut&> : T&
+typedef [T] Ref : TypeIf<T&, Not<IsRef<T>>>
+typedef [T] Ref : TypeIf<T, IsRef<T>>
```

Comment:
Does this result in a `Y&` if `T = Y mut&`?

Reply:
It would have. I've added a test and updated the traits namse to `ConstRef`, `MutRef`, and `IsAnyRef`